### DISCUSSION TITLE
Fixing Niko 552-72301 state

### DISCUSSION THREAD

**Discussion by VoltX** [Koenkk/zigbee2mqtt#20972] [Devices]
Hello everybody,

I looked into the issue https://github.com/Koenkk/zigbee2mqtt/issues/16645

Here are some test logs by modifying the "invert_cover" parameter, assuming that my shutter is physically closed:

- [invert_cover = true]
-- Received Zigbee message data '{"currentPositionLiftPercentage":0}'
-- Topic payload '{"linkquality":195,"position":0,"state":"OPEN"}'
- [invert_cover = false]
-- Received Zigbee message data '{"currentPositionLiftPercentage":0}'
-- Topic payload {"linkquality":195,"position":100,"state":"OPEN"}'

We see that the data received by Zigbee always gives me a position of 0. The "invert_cover" parameter inverts the MQTT result but the state remains the same "OPEN".

I parsed the definition in the niko.ts file and see that the fromZigbee property uses a "cover_position_tilt" converter. There are two possible parameters in this function `model.meta.coverInverted` and `options.invert_cover`: 
```js
const metaInvert = model.meta && model.meta.coverInverted;
const invert = metaInvert ? !options.invert_cover : options.invert_cover;
```
I found this a bit confusing, but it does appear that `options.invert_cover` inverts the position value returned by MQTT and `model.meta.coverInverted` allows the state to be inverted.

So I created an external converter to test my changes:
```js
const {} = require('zigbee-herdsman-converters/lib/modernExtend');
const fz = require('zigbee-herdsman-converters/converters/fromZigbee');
const tz = require('zigbee-herdsman-converters/converters/toZigbee');
const exposes = require('zigbee-herdsman-converters/lib/exposes');
const reporting = require('zigbee-herdsman-converters/lib/reporting');
const e = exposes.presets;

const definition = {
    zigbeeModel: ['Connectable motor control,3A'],
    model: '552-72301',
    vendor: 'Niko',
    description: 'Connectable motor control',
    fromZigbee: [fz.cover_position_tilt, fz.battery],
    toZigbee: [tz.cover_state, tz.cover_position_tilt],
    meta: {coverInverted:true},
    configure: async (device, coordinatorEndpoint, logger) => {
        const endpoint = device.getEndpoint(1);
        await reporting.bind(endpoint, coordinatorEndpoint, ['closuresWindowCovering']);
        await reporting.currentPositionLiftPercentage(endpoint);
    },
    exposes: [e.cover_position()],
};

module.exports = definition;
```

Simply adding the `meta: {coverInverted:true}` in the original definition (from niko.ts)

**And it works !**

![image](https://github.com/Koenkk/zigbee2mqtt/assets/3590072/b0d0d07f-e66a-4c40-a3c6-99214862374a)

It therefore seems that the parameter `options.invert_cover` should stay to **false** and **we just need to reverse the other parameter in the model.meta.coverInverted**.

I would like to tag the contributors to this part for information (from the issues https://github.com/Koenkk/zigbee2mqtt/issues/16645 https://github.com/Koenkk/zigbee2mqtt/issues/12755): @sjorge @bjorndeboodt @kjjdejong @MG2R

Can other people perform the test to confirm?

Can I create a pull-request based on this discussion to add this little line in the niko.ts definition?

**Comment by Koenkk** [comment]
> Can I create a pull-request based on this discussion to add this little line in the niko.ts definition?

Yes please!
